File: frontend/src/types/booking.ts
export type BookingStatus = 'pending' | 'approved' | 'rejected' | 'cancelled' | 'completed' | 'contract_ready' | 'contract_signed' | 'payment_pending'

// Matches backend: app/schemas/booking.py BookingRead
export interface Booking {
  id: number
  tenant_id: number
  property_id: number
  landlord_id: number
  status: BookingStatus
  message: string | null
  scheduled_date: string | null
  deposit_amount: number | null
  service_fee: number | null
  deposit_status: string | null
  payment_transaction_id: string | null
  lease_months: number | null
  total_rent: number | null
  application_data: any | null
  created_at: string
  updated_at: string
}

export interface BookingCreate {
  property_id: number
  message?: string
  scheduled_date?: string
  deposit_amount?: number
  service_fee?: number
  lease_months?: number
  total_rent?: number
  application_data?: any
}

export type NotificationType =
  | 'booking_created' | 'booking_approved' | 'booking_rejected' | 'booking_cancelled' | 'booking_completed'
  | 'payment_created' | 'payment_received' | 'payment_failed' | 'payment_expired'
  | 'contract_generated' | 'contract_signed' | 'system'

export interface Notification {
  id: number
  user_id: number
  type: NotificationType
  title: string
  content: string | null
  body?: string | null
  entity_type?: string | null
  entity_id?: string | null
  order_id?: string | null
  agreement_id?: string | null
  property_id?: number | null
  can_pay?: boolean
  payment_status?: string | null
  order_status?: string | null
  is_read: boolean
  created_at: string
  updated_at: string
}

export interface UnreadCount {
  count: number
}

File: backend/app/schemas/booking.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.booking import BookingStatus


class BookingCreate(BaseModel):
    unit_type_id: int
    institute_id: int | None = None
    tenant_id: int | None = None
    message: str | None = Field(default=None, max_length=2000)
    scheduled_date: str | None = Field(default=None, max_length=32)
    room_number: str | None = Field(default=None, max_length=20)
    deposit_amount: int | None = None
    service_fee: int | None = None
    lease_months: int | None = None
    total_rent: int | None = None
    application_data: dict | None = None


class BookingUpdate(BaseModel):
    status: BookingStatus | None = None
    deposit_status: str | None = None
    payment_transaction_id: str | None = None
    room_number: str | None = None
    contract_start: date | None = None
    contract_end: date | None = None


class BookingRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    tenant_id: int | None = None
    unit_type_id: int | None = None
    institute_id: int | None = None
    bm_id: int | None = None
    room_number: str | None = None
    status: BookingStatus
    message: str | None = None
    scheduled_date: str | None = None
    contract_start: date | None = None
    contract_end: date | None = None
    deposit_amount: int | None = None
    service_fee: int | None = None
    deposit_status: str | None = None
    payment_transaction_id: str | None = None
    lease_months: int | None = None
    total_rent: int | None = None
    application_data: dict | None = None
    created_at: datetime
    updated_at: datetime